### DISCUSSION TITLE
/the-basics/request.html

### DISCUSSION THREAD

**Discussion by giscus** [goravel/goravel#464] [Announcements]
# /the-basics/request.html

HTTP Requests Introduction The contracts/http/Request method of Goravel can interact with the current HTTP request processed by the application, and get the input and files subm...

https://www.goravel.dev/the-basics/request.html

**Comment by jxgame** [comment]
前端传来数组参数如何获取呀
···
page: 1
pageSize: 15
by_order_status: desc
by_order: id
search_date[0]: 2024-10-01 00:00:00
search_date[1]: 2024-11-30 00:00:00
status: 0
search_keyword: 133
···

上面的 search_date 参数获取不到，使用Input / InputArray /InputMap 都取不到，为空。求教。

**Reply by jxgame** [reply]
目前只能通过 Input("search_date[0]") 取得，但这样不优雅，也无法处理过多或不定长的数组呀。

**Reply by hwbrzzl** [reply]
上面链接可以改为：**?page=1&search_date=2024-10-01 00:00:00&search_date=2024-11-30 00:00:00&status=0....

不要带 [0] 这样的后缀，然后使用 QueryArray 可以正常获取。

**Comment by hpwm** [comment]
自定义Recover应该是这样，不然会一直提示错误
contractshttp "github.com/goravel/framework/contracts/http"
facades.Route().Recover(func(ctx contractshttp.Context, err any) {
		ctx.Request().AbortWithStatus(contractshttp.StatusInternalServerError)
	})

**Reply by hwbrzzl** [reply]
Both `ctx.Request().AbortWithStatus(contractshttp.StatusInternalServerError)` or `ctx.Response().String(500, "Internal Server Error").Abort()` should be fine.

**Comment by omiyz** [comment]
ctx.Request.All()为什么会将同一个字段的的不同值用","拼接哦, debug看到date原始值明明就是一个[]string,  用你的门面facades或只使用ctx.Request.Bind()方法永远也绑定不了query中的数组,  只有使用BindQuery才能正确绑定

**Reply by hwbrzzl** [reply]
请关注 issue https://github.com/goravel/goravel/issues/639

**Comment by merouanemc** [comment]
How Can I get the Raw body content ?

**Reply by hwbrzzl** [reply]
You can get the origin request via `ctx.Request().Origin()`, then you can get the Raw body easily.

**Comment by rowjak** [comment]
how to handle multiple upload files?
like example, in laravel i used 
        foreach ($request->file('images') as $image) {
            $path = $image->store('images', 'public'); // Store the image
            // You can store the $path in your database or use it for further processing
        }

**Reply by hwbrzzl** [reply]
We add a new function `ctx.Request().Files(name string) ([]filesystem.File, error)` in v1.16, it will be released soon.

**Comment by merouanemc** [comment]
Hi,

how can I cancel the request (stop querying dbDB do other instructions) when the user interrupts his request (Close browse, Hit Escape...), I pass the same http context to the db query but even that I close interrupt the request the app still querying the DB

Thanks

**Reply by hwbrzzl** [reply]
You can try to pass `ctx` into `Orm`: `facades.Orm().WithContext(ctx)`.

**Reply by merouanemc** [reply]
I did pass the context but still querying the db

**Reply by hwbrzzl** [reply]
Could you create a PR in [goravel/example](https://github.com/goravel/example)? I can check that.
